### PR TITLE
Add add all definitions button and multi drag and drop

### LINKED ISSUE #642: [Improvement]: Allow all columns of an entity
### Improvement description

Hi

Not sure if this is already possible but couldn't find proper docs about it. 
**Is it possible to allow all collumns/properties of an entity in the datahub without selecting them all via the interface?**

config:

`            schema:
                queryEntities:
                    Car:
                        id: Car
                        name: Car
                        columnConfig:
                            columns:
                                -
                                    attributes:
                                        attribute: sku
                                        label: Sku
                                        dataType: input
                                    isOperator: false
                                -
                                    attributes:
                                        attribute: workingTitle
                                        label: 'Working Title'
                                        dataType: input
                                    isOperator: false
                                -
                                    attributes:
                                        attribute: description
                                        label: Description
                                        dataType: wysiwyg
                                    isOperator: false`

### PR DESCRIPTION
**Pull request by noahcommandeur-twocream**
## Changes in this pull request  
Added a button that adds all definitions except for the system columns.
Furthermore, multi drag and drop has been added with STR + click you can select several fields and drag them over to the configuration.

You can deselect multiple fields and remove them from the configuration again

Button for delete complete configuration fixed

And category selector added ( STR + click ) on a category

## Additional info
https://github.com/user-attachments/assets/e494d937-f23e-4015-ae0a-7e57df561ad4

### CODE REVIEW

File: src/Resources/public/js/fieldConfigDialog.js [hunk 1]
```diff
@@ -54,13 +55,20 @@ pimcore.plugin.datahub.fieldConfigDialog = Class.create({
...
 
         buttons.push({
-                text: t("apply"),
-                iconCls: "pimcore_icon_apply",
-                handler: function () {
-                    this.commitData();
-                }.bind(this)
-            }
-        );
+            text: t("plugin_pimcore_datahub_add_all_definitions"),
```
Comment (blankse) [inline src/Resources/public/js/fieldConfigDialog.js:58]: The `plugin_pimcore_datahub_add_all_definitions` translation is missing.
image

**Comment by Nxooah** [comment]
Ouh sry I forgot to commit on 08.08 I'm back from vacation then I'll take care of it

File: src/Resources/public/js/fieldConfigDialog.js (hunk 1 above)
Comment (noahcommandeur-twocream) [reply src/Resources/public/js/fieldConfigDialog.js:58]: Done.

**Comment by mattamon** [comment]
Hey @noahcommandeur-twocream,

First of all, thank you for your PR! The functionality in the video looks really handy.

One minor thing for your PR  though. You do not need to change all the translation files. The English version is enough.
The translation get synced to POEditor https://poeditor.com/ get translated there and we get the translations back as a PR. 
Could you please remove all the other translations? :)

Thank you in advance!

**Comment by noahcommandeur-twocream** [comment]
Good morning @mattamon , 

All right, thanks for the feedback :)
In the future I will know that I only have to take care of the English translation
Is adapted!

Greetings Naoh

**Review by mattamon — CHANGES_REQUESTED** [review]
@noahcommandeur-twocream

I just tested it and it works like a charm!
It is a really cool feature! I like it a lot! 

https://github.com/pimcore/data-hub/blob/1.x/doc/01_Installation_and_Upgrade/01_Upgrade_Notes.md -> would you mind, adding it to the upgrade notes to 1.8? I think we should mention this specifically :) 

Some thoughts maybe. The `itemclick` handler could be its own function.
https://github.com/pimcore/data-hub/pull/871/files#diff-dbabce29090ce7119c18917a183b4fb8d7bb2eaacb6b287ff321efa2ecbd1645R520-R550 -> maybe this if else construct could be a bit simplified. 
My thougths can also be done in a separate PR of course.

**Comment by noahcommandeur-twocream** [comment]
Hi, thanks for the feedback, 
I'll take another look at the item click function and optimize it, then I'll expand the upgrade notes :)

Regards Noah

**Comment by mattamon** [comment]
> Hi, thanks for the feedback, I'll take another look at the item click function and optimize it, then I'll expand the upgrade notes :)
> 
> Regards Noah

Thank you Noah! Appreciate the effort you are putting in!

**Comment by Corepex** [comment]
@noahcommandeur-twocream just saw this PR - fantastic idea to add this button.

Are there any updates? I would love to put it into today's release :man_astronaut: :rocket:

**Comment by Nxooah** [comment]
Unfortunately I haven't quite got around to it yet, but I'll manage it in the next few days, maybe even today in the course of the day...
otherwise as a FollowUP during the days?

**Comment by Corepex** [comment]
> otherwise as a FollowUP during the days?

@noahcommandeur-twocream sure. You can also create a follow-up PR.
Do you have a few minutes to tell the world about your improvement in the 1.8 upgrade notes?

:man_astronaut: :rocket:

**Comment by Nxooah** [comment]
Well then I will add a code cleanup as a follow up :) I have opened a PR for the changes to the upgrade notes.
if you want to refine the text a little more you are welcome to do so :)
https://github.com/pimcore/data-hub/pull/884

**Comment by Corepex** [comment]
@noahcommandeur-twocream Thanks for the quick reply - now this PR can be part of the 1.8 release :100:  :1st_place_medal:

**Pull request merged**